src/core/evaluator.js

```diff
@@ -2656,20 +2654,26 @@ class PartialEvaluator {
       textContentItem.str.length = 0;
     }
 
-    function enqueueChunk() {
+    function enqueueChunk(batch = false) {
       const length = textContent.items.length;
-      if (length > 0) {
-        sink.enqueue(textContent, length);
-        textContent.items = [];
-        textContent.styles = Object.create(null);
+      if (length === 0) {
+        return;
+      }
+      if (batch && length < 10) {
```

Comment:
I'd prefer we make this 10 a `TEXT_CHUNK_BATCH_SIZE` constant at the top of the file with a short explanation of the value as a comment there, basically a small extract from the nice commit message that briefly explains why it should not be set too large.

Reply:
That's a *very* fair comment, and something that I should probably have done from the start; I'll try to come up with a reasonable (and not too long-winded comment).

Reply:
I've added a comment; please check that it makes sense and that it clearly conveys why the constant shouldn't be changed lightly!